### ISSUE TITLE
missing warning on possibly unbound variable

### ISSUE THREAD

**Issue by marcofrasvda** [astral-sh/ty#3894]
### Summary

This script always fails but no warning is given by `ty`. Pyrefly and basedpyright report the error as expected:
```py
def has_db() -> bool:
    return False

def load_settings() -> None:
    if has_db():
        doc = {"id": "abcde12345"}
    print(doc)
    # UnboundLocalError: cannot access local variable 'doc' where it is not associated with a value

load_settings()
```

I saw https://github.com/astral-sh/ty/issues/406 but I believe this case is simpler to address.
I saw https://github.com/astral-sh/ruff/issues/15916 but it's for Ruff and it's closed...

### Version

0.0.54

**Comment by sharkdp** [comment]
To catch errors like this, you need to turn on ty's [`possibly-unresolved-reference`](https://docs.astral.sh/ty/reference/rules/#possibly-unresolved-reference) rule.

[Playground](https://play.ty.dev/96c37c19-3acb-4a74-b201-c57632c39709)

**Comment by marcofrasvda** [comment]
Thanks for the information, I didn't realize this was an opt-in rule. For posterity, according to [this comment](https://github.com/astral-sh/ty/issues/555#issuecomment-2923384925) and the following one, this is currently disabled because it causes too many false positives, but hopefully things will improve before the stable release.

**Comment by sharkdp** [comment]
> hopefully things will improve before the stable release

We've found that a rule like this is generally prone to false positives. We do recommend turning it on if you want rather strict type checking, but we currently don't plan on improving anything on the road to stable (we're not aware of any specific things that we could do to make it better). So I wouldn't count on any major changes around this rule for stable.

**Comment by marcofrasvda** [comment]
I was quoting what I found in the linked comments, sad to hear that, I'll keep it on then since it's very valuable in preventing bugs. Thanks for the clarification.

**Comment by sharkdp** [comment]
In case you haven't seen it, you might be interested in this section in our docs: https://docs.astral.sh/ty/coming-from-mypy-or-pyright/#stricter-checking-with-ty
